### PR TITLE
[MI-3521] Fixed the issue of emojis showing as html from Teams to MM

### LINKED ISSUE #309: Emoji being sent from Teams->MM showing up with html
It appears that the emoji content generated from the suggested response buttons are different

Image

was generated using similar buttons as shown here:

Image

### PR DESCRIPTION
**Pull request by manojmalik20**
#### Summary
- Modified the msgToPost function to add <p> tag if missing from message
- Fixed the failing unit test of converters file
- Modified the order of checking duplicate post in the handleCreatedActivity function

#### Ticket Link
Fixes #309

### CODE REVIEW

**Review by ayusht2810 — COMMENTED** [review]
please fix ci

**Review by ayusht2810 — APPROVED** [review]

File: server/handlers/handlers.go [hunk 1]
```diff
@@ -174,6 +174,14 @@ func (ah *ActivityHandler) handleCreatedActivity(activityIds msteams.ActivityIds
 		return
 	}
 
+	// Avoid possible duplication
+	postInfo, _ := ah.plugin.GetStore().GetPostInfoByMSTeamsID(msg.ChatID+msg.ChannelID, msg.ID)
```
Comment (avas27JTG) [inline server/handlers/handlers.go:178]:
```suggestion
	postInfo, _ := ah.plugin.GetStore().GetPostInfoByMSTeamsID(msg.ChatID + msg.ChannelID, msg.ID)
```

**Review by avas27JTG — APPROVED** [review]

**Review by AayushChaudhary0001 — APPROVED** [review]

File: server/handlers/handlers.go [hunk 2]
```diff
@@ -174,6 +174,14 @@ func (ah *ActivityHandler) handleCreatedActivity(activityIds msteams.ActivityIds
...
+	if postInfo != nil {
+		ah.plugin.GetAPI().LogDebug("duplicate post")
```
Comment (Kshitij-Katiyar) [inline server/handlers/handlers.go:180]: Should we add additional informations like msg.ID here in the Log ?

**Review by Kshitij-Katiyar — APPROVED** [review]

File: server/handlers/handlers.go (hunk 1 above)
Comment (manojmalik20) [reply server/handlers/handlers.go:178]: The linter doesn't allow this.

File: server/handlers/handlers.go (hunk 2 above)
Comment (manojmalik20) [reply server/handlers/handlers.go:180]: I don't think there's any use for that.

File: server/handlers/converters.go [hunk 3]
```diff
@@ -38,6 +38,9 @@ func (ah *ActivityHandler) msgToPost(channelID, senderID string, msg *msteams.Me
 	var embeddedImages []msteams.Attachment
 	text, embeddedImages = ah.handleImages(text)
 	msg.Attachments = append(msg.Attachments, embeddedImages...)
+	if !strings.HasPrefix(text, "<p>") {
```
Comment (jespino) [inline server/handlers/converters.go:41]: I think a better way to do it to put it as an exception in the `ConvertToMD`  function, that has a check at the beginning for text that contains `<div>` or  `<p>`, you can add `<img ` or any other chunk of text that clearly means that there is some HTML to be process.

**Review by jespino — COMMENTED** [review]
An improvement about how to handle the case here

**Pull request merged**